### ISSUE TITLE
[spaceship] Creation of in-app purchases returns NotFound error

### ISSUE THREAD

**Issue by philippmachs** [fastlane/fastlane#19140]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
Creation of in-app purchases via Spaceship::Tunes client returns Not Found error even after the fix introduced in 2.188.0

##### Command executed

  <pre>
APPLE_USER=*** APPLE_PASS=*** APPLE_ID=*** VERBOSE=true bundle exec irb 
require 'spaceship'
Spaceship::Tunes::login(ENV['APPLE_USER'],ENV['APPLE_PASS'])
Spaceship::Tunes::Application.find(ENV['APPLE_ID']).in_app_purchases.create!({"type":"nonConsumable","versions":{"en-US":{"name":"test name","description":"test desctiption"}},"reference_name":"test refname","product_id":"1230987","review_notes":"","pricing_intervals":[{"country":"WW","tier":550}]})
</pre>

##### Complete output when running fastlane, including the stack trace and command used

<details>

  <pre> 
# INFO  [12:39:32]: >> GET ra/apps/manageyourapps/summary/v2: [undefined body]
# DEBUG [12:39:33]: << GET ra/apps/manageyourapps/summary/v2: 200 {...}
# INFO  [12:39:40]: >> GET ra/apps/***/iaps/nonConsumable/template: [undefined body]
# DEBUG [12:39:41]: << GET ra/apps/***/iaps/nonConsumable/template: 200 {...}
# INFO  [12:39:41]: >> POST ra/apps/***/iaps:
# DEBUG [12:39:41]: << POST ra/apps/***/iaps: 404 {"data"=>nil, "messages"=>{"warn"=>nil, "error"=>["Not Found"], "info"=>nil}, "statusCode"=>"ERROR"}
# Traceback (most recent call last):
#        16: from .../ruby-2.7.0/lib/ruby/2.7.0/bundler/vendor/thor/lib/thor/base.rb:476:in `start'
#        15: from .../ruby-2.7.0/lib/ruby/2.7.0/bundler/cli.rb:30:in `dispatch'
#        14: from .../ruby-2.7.0/lib/ruby/2.7.0/bundler/vendor/thor/lib/thor.rb:399:in `dispatch'
#        13: from .../ruby-2.7.0/lib/ruby/2.7.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
#        12: from .../ruby-2.7.0/lib/ruby/2.7.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
#        11: from .../ruby-2.7.0/lib/ruby/2.7.0/bundler/cli.rb:476:in `exec'
#        10: from .../ruby-2.7.0/lib/ruby/2.7.0/bundler/cli/exec.rb:28:in `run'
#         9: from .../ruby-2.7.0/lib/ruby/2.7.0/bundler/cli/exec.rb:63:in `kernel_load'
#         8: from .../ruby-2.7.0/lib/ruby/2.7.0/bundler/cli/exec.rb:63:in `load'
#         7: from .../ruby-2.7.0/bin/irb:23:in `<top (required)>'
#         6: from .../ruby-2.7.0/bin/irb:23:in `load'
#         5: from .../ruby-2.7.0/lib/ruby/gems/2.7.0/gems/irb-1.2.1/exe/irb:11:in `<top (required)>'
#         4: from (irb):8
#         3: from .../ruby-2.7.0/gems/fastlane-2.188.0/spaceship/lib/spaceship/tunes/iap.rb:76:in `create!'
#         2: from .../ruby-2.7.0/gems/fastlane-2.188.0/spaceship/lib/spaceship/tunes/tunes_client.rb:1422:in `create_iap!'
#         1: from .../ruby-2.7.0/gems/fastlane-2.188.0/spaceship/lib/spaceship/tunes/tunes_client.rb:230:in `handle_itc_response'
# Spaceship::Tunes::Error (Not Found)
</pre>
</details>

### Environment

<details>

### Stack

| Key                         | Value                                |
| --------------------------- | ------------------------------------ |
| OS                          | 10.15.6                              |
| Ruby                        | 2.7.0                                |
| Bundler?                    | true                                 |
| Git                         | git version 2.28.0                   |
| Installation Source         | ~/.rvm/gems/ruby-2.7.0/bin/fastlane  |
| Host                        | Mac OS X 10.15.6 (19G2021)           |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.7.0/lib         |
| OpenSSL Version             | OpenSSL 1.1.1i  8 Dec 2020           |
| Is contained                | false                                |
| Is homebrew                 | false                                |
| Is installed via Fabric.app | false                                |
| Xcode Path                  | /Library/Developer/CommandLineTools/ |
| Xcode Version               |                                      |
| Swift Version               | 5.3                                  |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

default_platform(:ios)

platform :ios do
  desc "Description of what the lane does"
  lane :custom_lane do
    # add actions here: https://docs.fastlane.tools/actions
  end
end

```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.188.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                             | Version      |
| ------------------------------- | ------------ |
| did_you_mean                    | 1.4.0        |
| rake                            | 13.0.6       |
| CFPropertyList                  | 3.0.3        |
| public_suffix                   | 4.0.6        |
| addressable                     | 2.8.0        |
| artifactory                     | 3.0.15       |
| atomos                          | 0.1.3        |
| aws-eventstream                 | 1.1.1        |
| aws-partitions                  | 1.479.0      |
| aws-sigv4                       | 1.2.4        |
| jmespath                        | 1.4.0        |
| aws-sdk-core                    | 3.117.0      |
| aws-sdk-kms                     | 1.44.0       |
| aws-sdk-s3                      | 1.96.2       |
| babosa                          | 1.0.4        |
| bundler                         | 2.1.2        |
| claide                          | 1.0.3        |
| colored                         | 1.2          |
| colored2                        | 3.1.2        |
| highline                        | 2.0.3        |
| commander                       | 4.6.0        |
| declarative                     | 0.0.20       |
| digest-crc                      | 0.6.3        |
| unf_ext                         | 0.0.7.7      |
| unf                             | 0.1.4        |
| domain_name                     | 0.5.20190701 |
| dotenv                          | 2.7.6        |
| emoji_regex                     | 3.2.2        |
| excon                           | 0.85.0       |
| faraday-em_http                 | 1.0.0        |
| faraday-em_synchrony            | 1.0.0        |
| faraday-excon                   | 1.1.0        |
| faraday-httpclient              | 1.0.1        |
| faraday-net_http                | 1.0.1        |
| faraday-net_http_persistent     | 1.2.0        |
| faraday-patron                  | 1.0.0        |
| multipart-post                  | 2.0.0        |
| ruby2_keywords                  | 0.0.5        |
| faraday                         | 1.5.1        |
| http-cookie                     | 1.0.4        |
| faraday-cookie_jar              | 0.0.7        |
| faraday_middleware              | 1.0.0        |
| fastimage                       | 2.2.4        |
| gh_inspector                    | 1.1.3        |
| jwt                             | 2.2.3        |
| memoist                         | 0.16.2       |
| multi_json                      | 1.15.0       |
| os                              | 1.1.1        |
| signet                          | 0.15.0       |
| googleauth                      | 0.16.2       |
| httpclient                      | 2.8.3        |
| mini_mime                       | 1.1.0        |
| trailblazer-option              | 0.1.1        |
| uber                            | 0.1.0        |
| representable                   | 3.1.1        |
| retriable                       | 3.1.2        |
| rexml                           | 3.2.5        |
| webrick                         | 1.7.0        |
| google-apis-core                | 0.4.1        |
| google-apis-androidpublisher_v3 | 0.8.0        |
| google-apis-playcustomapp_v1    | 0.5.0        |
| google-apis-iamcredentials_v1   | 0.6.0        |
| google-apis-storage_v1          | 0.6.0        |
| google-cloud-env                | 1.5.0        |
| google-cloud-errors             | 1.1.0        |
| google-cloud-core               | 1.6.0        |
| google-cloud-storage            | 1.34.1       |
| json                            | 2.5.1        |
| mini_magick                     | 4.11.0       |
| naturally                       | 2.2.1        |
| plist                           | 3.6.0        |
| rubyzip                         | 2.3.2        |
| security                        | 0.1.3        |
| simctl                          | 1.6.8        |
| terminal-notifier               | 2.0.0        |
| unicode-display_width           | 1.7.0        |
| terminal-table                  | 1.8.0        |
| tty-screen                      | 0.8.1        |
| tty-cursor                      | 0.7.1        |
| tty-spinner                     | 0.9.3        |
| word_wrap                       | 1.0.0        |
| nanaimo                         | 0.3.0        |
| xcodeproj                       | 1.20.0       |
| rouge                           | 2.0.7        |
| xcpretty                        | 0.3.0        |
| xcpretty-travis-formatter       | 1.0.1        |
</details>

</details>

**Comment by hziKen** [comment]
Have you fixed this problem

**Comment by philippmachs** [comment]
Works like a charm. Thanks a lot!

**Comment by carolecho** [comment]
> Works like a charm. Thanks a lot!
 Add x-csrf-itc header to requests #19145  ?

**Comment by philippmachs** [comment]
Yes

**Comment by pfay-fc** [comment]
Same issue on editing (and this comment confirms the fix for that case as well) https://github.com/fastlane/fastlane/pull/19145#pullrequestreview-715675727

**Comment by jmei08** [comment]
It works. Thanks #19145

**Comment by maurovz** [comment]
I'm still having this issue as of `2.189.0`

**Comment by somuch** [comment]
> I'm still having this issue as of `2.189.0`

They do weekly releases to Ruby Gem. The same as you, I am also waiting for the new release. I believe we can see the release in about 2 or 3 days.

**Comment by luzongyuan** [comment]
today ，I meet it ,too

**Comment by hziKen** [comment]
update fastlane to 2.190

**Comment by lacostej** [comment]
I believe this was solved by #19145
